src/app/onboard/member/page.tsx

```diff
@@ -10,10 +11,37 @@ import { useOnboardingDetailsStore } from "@/stores/useOnboardingDetailsStore";
 const MembershipType = () => {
   const member = useOnboardingDetailsStore((state) => state.member);
   const setMember = useOnboardingDetailsStore((state) => state.setMember);
+  const firstName = useOnboardingDetailsStore((state) => state.firstName);
+  const lastName = useOnboardingDetailsStore((state) => state.lastName);
   const router = useRouter();
+  const isNameProvided =
+    (firstName !== null && lastName !== null) ||
+    (firstName !== "" && lastName !== "") ||
+    (!!firstName && !!lastName);
+  if (!isNameProvided) {
+    router.push("onboard/name");
+  }
```

Comment:
you can simplify the former

you won't be able to use router push before the page loads so you can use redirect isntead from "next/navigation"